src/Features/Core/Portable/MetadataAsSource/MetadataAsSourceOptions.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.MetadataAsSource
+{
+    /// <summary>
+    /// Options for metadata as source navigation
+    /// </summary>
+    /// <param name="NavigateToDecompiledSources"><see langword="false"/> to disallow decompiling code, which may
+    /// result in signagures only being returned if there is no other non-decompilation option available</param>
+    /// <param name="AlwaysUseDefaultSymbolServers">Whether navigation should try to use the default Microsoft and
```

Comment:
do you have an example scenario of when someone would want to actually turn off this option?  Presumably if there is a better match in a custom symbol server we'd use that anyway

Reply:
I think it's literally for "we don't want network traffic going to Microsoft/Nuget that reveals what we're doing" case